### PR TITLE
Rdbc 171 command unify [WIP]

### PR DESCRIPTION
**Pull request by dabroz**

### CODE REVIEW

File: lib/ravendb/documents/commands/hilo_next_command.rb [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
       @tag = tag
       @last_batch_size = last_batch_size
       @last_range_at = last_range_at
       @last_range_max = last_range_max
       @identity_parts_separator = identity_parts_separator
     end
 
-    def create_request(server_node)
-      @params = {
+    def create_request(server_node, url:)
```
Comment (gregolsky) [inline lib/ravendb/documents/commands/hilo_next_command.rb]: Just FYI I dropped the URL output parameter from `createRequest()` in Node.js client. It appears it's not used anywhere. I think you can safely remove it as well

File: lib/ravendb/http/raven_command.rb [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      @params = params
+      @payload = payload
+      @headers = headers
+      @failed_nodes = Set.new([])
+      @_last_response = nil
+      @mapper = JsonObjectMapper.new
+      @response_type = :object
+    end
+
+    def can_cache?
```
Comment (gregolsky) [inline lib/ravendb/http/raven_command.rb:22]: Please add `can_cache_aggresively` https://github.com/ravendb/ravendb-jvm-client/blob/48d6516935eb78009e000b6f72687318d7fb088f/src/main/java/net/ravendb/client/http/RavenCommand.java#L58

Also `can_cache` and `can_cache_agressively` are set to `true` in ctor.

File: lib/ravendb/http/raven_command.rb [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        @headers.each do |header, value|
+          request.add_field(header, value)
+        end
+      end
+
+      request
+    end
+
+    def set_response(response)
+      @_last_response = response
```
Comment (gregolsky) [inline lib/ravendb/http/raven_command.rb:78]: Logic in JVM client seems to be a bit different https://github.com/ravendb/ravendb-jvm-client/blob/48d6516935eb78009e000b6f72687318d7fb088f/src/main/java/net/ravendb/client/http/RavenCommand.java#L72

**Comment by gregolsky** [comment]
HTTPS tests failed for some reason. https://travis-ci.org/ravendb/ravendb-ruby-client/builds/381149815?utm_source=github_status&utm_medium=notification

**Comment by gregolsky** [comment]
Also we'd like to port operations as well - it is the operations that are going to be used by the user, not commands directly.

File: lib/ravendb/documents/commands/hilo_next_command.rb (hunk 1 above)
Comment (dabroz) [reply lib/ravendb/documents/commands/hilo_next_command.rb]: Done.

**Pull request merged**